### PR TITLE
fix tag image

### PR DESCRIPTION
**Pull request by jimmyxian**
Fix: https://github.com/docker/swarm/issues/1149
Support `tag image` for all matched images
Signed-off-by: Xian Chaobo <xianchaobo@huawei.com>

### CODE REVIEW

**Comment by jimmyxian** [comment]
ping @vieux @tchk1228

**Comment by vieux** [comment]
@jimmyxian why did you need to change the test ?

**Comment by jimmyxian** [comment]
@vieux The original test just support "one node with one image" case, can not cover "multi-nodes with same image" case. So I mix two cases.

**Comment by chanwit** [comment]
@jimmyxian It's a good practice to not changing the tests. 

Could you please just adding its as a new test, say "docker tag - multi-nodes" or something similar? With this, we can ensure that your PR does not break the old ones.

**Comment by jimmyxian** [comment]
@chanwit 
Agree. I will modify it. :)

**Comment by jimmyxian** [comment]
ping @vieux @chanwit 
Review Again. :)

**Comment by chanwit** [comment]
LGTM

File: test/integration/api/tag.bats [hunk 1]
```diff
@@ -30,3 +30,22 @@ function teardown() {
 	[ "$status" -eq 0 ]
 	[[ "${output}" == *"tag_busybox"* ]]
 }
+
+@test "docker tag multi-nodes with same image" {
+	start_docker_with_busybox 2
+	swarm_manage
+
+	# make sure the image of busybox exists
+	# and tag_busybox not exsit
```
Comment (chanwit) [inline test/integration/api/tag.bats]: woah typo here @jimmyxian

Comment (jimmyxian) [inline test/integration/api/tag.bats]: Ehm..Right...

**Comment by vieux** [comment]
LGTM

**Pull request merged**